### PR TITLE
Add target to rdd data, update command name and name for RDD

### PR DESCRIPTION
**Pull request by kartang**

### CODE REVIEW

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts [hunk 1]
```diff
@@ -1454,8 +1454,10 @@ class AppInsightsTests extends TestClass {
             test: () => {
                 var appInsights = new Microsoft.ApplicationInsights.AppInsights(this.getAppInsightsSnippet());
                 var trackStub = this.sandbox.stub(appInsights.context, "track");
-                var name = "test";
-                var url = "http://myurl.com";
+                var pathName = "https://tempurl.net/api/temp/ABCD";
+                var commandName = "GET https://tempurl.net/api/temp/ABCD?param1=test&param2=test";
+                var url = "https://tempurl.net/api/temp/ABCD?param1=test&param2=test";
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: It doesn't really matter but I would probably have built url as pathName + the param string and then build commandName as "GET " + url so there would only be one copy of the string to type/maintain.

Comment (kartang) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: Makes sense. Will update that. Thanks!

File: JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts [hunk 2]
```diff
@@ -39,8 +40,10 @@ module Microsoft.ApplicationInsights.Telemetry {
             super();
 
             this.id = id;
-            this.name = this.formatDependencyName(method, absoluteUrl);
-            this.commandName = Common.DataSanitizer.sanitizeUrl(commandName);
+            this.target = UrlHelper.parseUrl(absoluteUrl).hostname;
+            this.commandName = this.formatDependencyName(method, absoluteUrl);
```
Comment (OsvaldoRosado) [inline JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts]: What about the `data` field that replaces `commandName`?

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts [hunk 3]
```diff
@@ -1607,8 +1611,9 @@ class AppInsightsTests extends TestClass {
                 // Assert
                 Assert.ok(trackStub.called, "Track should be called");
                 var rdd = <Microsoft.ApplicationInsights.Telemetry.RemoteDependencyData>(<any>trackStub.args[0][0]).data.baseData;
-                Assert.equal(url, rdd.name);
-                Assert.equal(name, rdd.commandName);
+                Assert.equal(pathName, rdd.name);
+                Assert.equal(url, rdd.commandName);
```
Comment (OsvaldoRosado) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: Is this right? The url variable holds the plain url, shouldn't the commandName object have "GET "+url?

Comment (kartang) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: The method here is null, and in that case, url will be the command name.

File: JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts (hunk 2 above)
Comment (kartang) [inline JavaScript/JavaScriptSDK/Telemetry/RemoteDependencyData.ts]: I believe the transformation is handled in breeze? I though breeze today expects commandName  as opposed to data?

**Comment by kamilszostak** [comment]
@SergeyKanzhelev Updating schema might be a breaking change. I would suggest updating in in next minor release (1.1) - #327.

File: JavaScript/JavaScriptSDK/TelemetryContext.ts [hunk 4]
```diff
@@ -318,5 +320,67 @@ module Microsoft.ApplicationInsights {
...
+            try {
+                var data = (<any>envelope).data;
+                if (data && data.baseType === Microsoft.ApplicationInsights.Telemetry.RemoteDependencyData.dataType) {
+                    var rddData = <Microsoft.ApplicationInsights.Telemetry.RemoteDependencyData>(<any>data.baseData);
+                    if (rddData) {
+                        this._fixRDDDepricatedValues(rddData);
+                    }
+                }
+            } catch (e) {
+            }
```
Comment (kamilszostak) [inline JavaScript/JavaScriptSDK/TelemetryContext.ts]: Consider logging a warning message. It won't be reported unless a user is in the debug mode or has enabled verboseLogging. 

sample code: 
                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING,
                    new _InternalLogMessage(_InternalMessageId.<newEnumName>, "error message" + Util.getExceptionName(e),
                        { exception: Util.dump(e) }));

Comment (kartang) [inline JavaScript/JavaScriptSDK/TelemetryContext.ts]: Thanks! Updated the logging!

File: JavaScript/JavaScriptSDK/TelemetryContext.ts [hunk 5]
```diff
@@ -331,6 +331,9 @@ module Microsoft.ApplicationInsights {
                     }
                 }
             } catch (e) {
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING,
+                    new _InternalLogMessage(_InternalMessageId.NONUSRACT_FailedToFixDepricatedValues, "Failed to parse the base data object, to fix the depricated values " + Util.getExceptionName(e),
```
Comment (OsvaldoRosado) [inline JavaScript/JavaScriptSDK/TelemetryContext.ts]: Typo: Depricated -> Deprecated

File: JavaScript/JavaScriptSDK.Tests/JavaScriptSDK.Tests.csproj [hunk 6]
```diff
@@ -140,6 +140,9 @@
     <Content Include="E2ETests\standalone\sprint69.html" />
     <Content Include="E2ETests\standalone\test.html" />
     <Content Include="E2ETests\standalone\tests.js" />
+    <Content Include="geckodriver.exe">
```
Comment (dnduffy) [inline JavaScript/JavaScriptSDK.Tests/JavaScriptSDK.Tests.csproj]: Why is it "Content"?

**Pull request merged**